numpy/linalg/linalg.py

```diff
@@ -2376,34 +2377,36 @@ def multi_dot(arrays):
     Compute the dot product of two or more arrays in a single function call,
     while automatically selecting the fastest evaluation order.
 
-    `multi_dot` chains `numpy.dot` and uses optimal parenthesization
+    `multi_dot` chains `numpy.matmul` and uses optimal parenthesization
     of the matrices [1]_ [2]_. Depending on the shapes of the matrices,
     this can speed up the multiplication a lot.
 
     If the first argument is 1-D it is treated as a row vector.
     If the last argument is 1-D it is treated as a column vector.
-    The other arguments must be 2-D.
+    If one of the other arguments is N-D, N>2, it is treated as a stack of matrices and broadcast accordingly
```

Comment:
Does this mean that the first and last arguments can't be N-D? Why not? Couldn't these also be treated like stacks of matrices?